src/fluree/db/db/json_ld.cljc

```diff
@@ -442,6 +442,15 @@
         (vswap! context-cache assoc-in [context-type supplied-context] parsed-ctx)
         parsed-ctx)))
 
+(defn default-context-update
+  "Updates default context, so on next commit it will get written in the commit file."
+  [db default-context]
+  (let [default-context* (-> default-context
+                             (ctx-util/mapify-context (dbproto/-default-context db)) ;; allows 'extending' existing default context using empty string ""
+                             (ctx-util/stringify-context))]
+    (assoc db :default-context default-context*
+              :context-cache (volatile! {})
```

Comment:
It looks like this resets the context cache, right? Do we expect enough contention on updating an atom here that it's worth introducing the risk of a volatile? I only see one place where we update it currently, so it seems safe at the moment.

Reply:
It replaces the cache, not reset (which would have created issues). Any older versions of the DB which still have a handle on the old cache will continue to work as expected.

It is safe here at the moment, but it is possible future code could make it unsafe so worth considering. Performance is such a driving criteria for us, I opt towards the volatile - however it is possible a future modification could change if a volatile remains viable.

Reply:
Sorry, didn't address fully the second part of the question. The cache's results are pure, so while I expect update contention to be uncommon, when it does happen we don't care which thread wins and there is no dependent state. 

Here the cache is acting in a way that a `delay` function perhaps could have been used. keywordizing a small context isn't super concerning, but doing so for e.g. the entire schema.org context, which is massive, would require a lot of work that might never get used.